snap-preload.c

```diff
@@ -27,14 +27,14 @@ static int DEBUG = 0;
 #define log(FORMAT, ...) if (DEBUG) {fprintf(stderr, "snap-preload: " FORMAT "\n", __VA_ARGS__);}
 
 static char *adjust_shm_path(const char *orig_path) {
-  char *new_path = malloc(PATH_MAX);
+  int path_len = strlen(orig_path) + strlen(SNAP_INSTANCE_NAME) + 8;
+  char *new_path = malloc(path_len);
```

Comment:
add a `assert(new_path != NULL)`. In an OoM scenario you want to crash giving some hint about the reason.